qiskit/circuit/quantumcircuit.py

```diff
@@ -970,34 +988,40 @@ def compose(
         variable_mapper = _classical_resource_map.VariableMapper(
             dest.cregs, edge_map, dest.add_register
         )
-        mapped_instrs: CircuitData = CircuitData(dest.qubits, dest.clbits, reserve=len(other.data))
-        for instr in other.data:
-            n_qargs: list[Qubit] = [edge_map[qarg] for qarg in instr.qubits]
-            n_cargs: list[Clbit] = [edge_map[carg] for carg in instr.clbits]
-            n_op = instr.operation.copy()
+
+        def map_vars(op):
+            # TODO: do we really need to copy if we're not updating vars?
+            n_op = op.copy()
```

Comment:
For the most part, yes - what if some other method after this were to mutate it?  As we move more operations to be singletons, the cost of this goes down hugely anyway.